[stable/nginx-ingress] Add variable for allowPrivilegeEscalation

mblaschke opened the pull request:
#### What this PR does / why we need it:
Allows port 80 and 443 using NET_BIND_SERVICE as a non-root container this also needs privilegeEscalation to be enabled. The default value of `allowPrivilegeEscalation` can be `false` due to a `PodSecurityPolicy` so this value should be changed to true.

Otherweise the service cannot open port 80 and 443 due to a permission denied.

#### Which issue this PR fixes

#### Special notes for your reviewer:

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped

@mgoodness  @jackzampolin

ChiefAlexander commented on stable/nginx-ingress/templates/controller-daemonset.yaml:
```diff
@@ -85,6 +85,7 @@ spec:
                 add:
                 - NET_BIND_SERVICE
             runAsUser: {{ .Values.controller.image.runAsUser }}
+            allowPrivilegeEscalation: true
```
Instead of explicitly setting this value add it to the values file and pull it in here. This way people can set it true or false depending on their environment.

mblaschke replied:
Done :)

mblaschke committed:
Add allowPrivilegeEscalation as variable

Adds variable for allowPrivilegeEscalation which allows to use NET_BIND_SERVICE with a PodSecurityPolicy which forbids privilege escalation.

Signed-off-by: Markus Blaschke <mblaschke82@gmail.com>

taharah commented on stable/nginx-ingress/values.yaml:
```diff
@@ -9,6 +9,7 @@ controller:
     pullPolicy: IfNotPresent
     # www-data -> uid 33
     runAsUser: 33
+    allowPrivilegeEscalation: false
```
This needs to be set to `true` when binding to ports 80 and 443 as a non-root user since the default image has the capabilities set directly on the binaries (see [Dockerfile](https://github.com/kubernetes/ingress-nginx/blob/nginx-0.25.0/rootfs/Dockerfile#L49-L53) for reference) and Kubernetes does not currently support ambient capabilities (see [kubernetes/kubernetes#56374](https://github.com/kubernetes/kubernetes/issues/56374) for reference).

mblaschke replied:
@taharah done

mblaschke committed:
Set allowPrivilegeEscalation to true

Should be enabled as NET_BIND_SERVICE is active by default

Signed-off-by: Markus Blaschke <mblaschke82@gmail.com>
